src/analyses/mCPAccess.ml

```diff
@@ -0,0 +1,43 @@
+
+open MCPRegistry
+
+(** Access module corresponding to MCP.
+    Separate to avoid dependency cycle. *)
+module A =
+struct
+  open AccListSpec
+  open List
+  open Obj
+
+  include DomListPrintable (PrintableOfMCPASpec (AccListSpec))
+
+  let unop_fold f a (x:t) =
+    let f a n d =
+      f a n (assoc_dom n) d
+    in
+    fold_left (fun a (n,d) -> f a n d) a x
+
+  let binop_fold f a (x:t) (y:t) =
+    let f a n d1 d2 =
+      f a n (assoc_dom n) d1 d2
+    in
+    try if length x <> length y
+      then raise (DomListBroken "binop_fold : differing lengths")
+      else fold_left (fun a (n,d) -> f a n d @@ assoc n y) a x
+    with Not_found -> raise (DomListBroken "binop_fold : assoc failure")
+
+  let may_race x y = not @@ binop_fold (fun a n (module S: Analyses.MCPA) x y ->
+      a || not (S.may_race (obj x) (obj y))
+    ) false x y
```

Comment:
Is there some deeper meaning behind the double negation?

```suggestion
  let may_race x y = binop_fold (fun a n (module S: Analyses.MCPA) x y ->
      a && (S.may_race (obj x) (obj y))
    ) true x y
```

Reply:
At some point I had to insert the negations to get things to work, but I'm not sure anymore. It might've been because `conflict` was a confusing name, which implied nothing about may vs must. I'll check again if this can be simplified.